src/kuzya.cpp

```diff
@@ -801,9 +855,65 @@ void Kuzya::refreshProfileSettings()
 /**
 *******************************************************************************************************
 **/
+//bool Kuzya::slotSave(void)
+//{
+//    QString newFileName;
+//    if (fileName.isEmpty())
+//    {
+//        // fileName = fileDialog->getSaveFileName(this, tr("Save as..."), DefaultDir, filter, &currentFilter);
+//        refreshDialogSettings();
+//        fileDialog->setAcceptMode(QFileDialog::AcceptSave);
+//        fileDialog->setFileMode(QFileDialog::AnyFile);
+
+//        if (fileDialog->exec())
+//            newFileName = fileDialog->selectedFiles().at(0);
+
+//        if (newFileName.isEmpty()) return false;
+//        else fileName = newFileName;
+//    }
+
+
+//    QFile file(fileName);
+//    if (!file.open(QIODevice::WriteOnly | QIODevice::Text))
+//    {
+//        QMessageBox *msgBox= new QMessageBox();
+//        msgBox->setIcon(QMessageBox::Warning);
+//        msgBox->setWindowTitle(tr("File cannot be saved "));
+//        msgBox->setText(tr("Do you have permision to access data in this folder? Select another place to save this file"));
+//        QAbstractButton *OkBtn = msgBox->addButton(tr("Ok"),QMessageBox::ActionRole);
+//        msgBox->exec();
+//        if (msgBox->clickedButton()==(OkBtn))
+//        {
+//            //slotSave_as();
+//        }
+//        delete msgBox;
+//        return false ;
+//    }
+//    QTextStream stream(&file);
+//    stream << textEditor->text();
+//    textEditor->setModified(false);
+//    file.close();
+//    statusBar()->showMessage(tr("Saved"), 2000);
+//    addFileNameToList(file.fileName());
+//    refreshProfileSettings();
+//    slotUpdateWindowName(false);
+//    if(settings->isLineMarginVisible) textEditor->setMarginWidth(3,QVariant(textEditor->lines()).toString());
+//    return true;
+//}
+
+void Kuzya::slotBeforeSave(void)
```

Comment:
You could check for !myFile->isSaved() in slot save(). You do not need that slot Kuzya::slotBeforeSave(void)

Reply:
Moved the check for !myFile->isSaved() in slotSave and removed slotBeforeSave.